Sources/ProjectDescription/Settings.swift

```diff
@@ -124,12 +124,66 @@ public extension CustomConfiguration {
 /// The default settings can be overridden via `Settings base: SettingsDictionary`
 /// and `Configuration settings: SettingsDictionary`.
 ///
-/// - all: Essential settings plus all the recommended settings (including extra warnings)
+/// - `recommended`: Essential settings plus all the recommended settings (including extra warnings)
 /// - essential: Only essential settings to make the projects compile (i.e. `TARGETED_DEVICE_FAMILY`)
-public enum DefaultSettings: String, Codable {
-    case recommended
-    case essential
+public enum DefaultSettings: Codable, Equatable {
+    case recommended(excluding: Set<String> = [])
+    case essential(excluding: Set<String> = [])
     case none
+
+    enum CodingKeys: CodingKey {
+        case recommended, essential, none
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        let key = container.allKeys.first
+
+        switch key {
+        case .recommended:
+            var nestedContainer = try container.nestedUnkeyedContainer(forKey: .recommended)
+            let excludedKeys = try nestedContainer.decode(Set<String>.self)
+            self = .recommended(excluding: excludedKeys)
+        case .essential:
+            var nestedContainer = try container.nestedUnkeyedContainer(forKey: .essential)
+            let excludedKeys = try nestedContainer.decode(Set<String>.self)
+            self = .essential(excluding: excludedKeys)
+        case .none:
+            self = .none
+        default:
+            throw DecodingError.dataCorrupted(
+                DecodingError.Context(
+                    codingPath: container.codingPath,
+                    debugDescription: "Unable to decode DefaultSettings."
```

Comment:
We can make this error message a little bit more descriptive by writing out available cases